component/src/main/java/org/wso2/extension/siddhi/io/cdc/source/polling/strategies/WaitOnMissingRecordPollingStrategy.java

```diff
@@ -0,0 +1,176 @@
+                    while (resultSet.next()) {
+                        boolean isTimedout = false;
+                        int currentPollingColumnValue = resultSet.getInt(pollingColumn);
+                        if (currentPollingColumnValue - lastReadPollingColumnValue > 1) {
+                            if (waitingFor == -1) {
+                                // This is the first time to wait for the current record. Hence set the expected record
+                                // id and the current timestamp.
+                                waitingFor = lastReadPollingColumnValue + 1;
+                                waitingFrom = System.currentTimeMillis();
+                            }
+
+                            isTimedout = waitTimeout > -1 && waitingFrom + waitTimeout < System.currentTimeMillis();
+                            if (!isTimedout) {
+                                log.debug("Missing record found at " + waitingFor + ". Hence pausing the process and " +
+                                        "retry in " + pollingInterval + " seconds.");
+                                breakOnMissingRecord = true;
+                                break;
+                            }
+                        }
+                        if (waitingFor > -1) {
+                            if (isTimedout) {
+                                log.debug("Waiting for missed record " + waitingFor + " timed-out. Hence resuming " +
+                                        "the process.");
+                            } else {
+                                log.debug("Received the missed record " + waitingFor + ". Hence resuming the process.");
+                            }
+                            waitingFor = -1;
+                            waitingFrom = -1;
+                        }
+                        detailsMap = new HashMap<>();
+                        for (int i = 1; i <= metadata.getColumnCount(); i++) {
+                            String key = metadata.getColumnName(i);
+                            Object value = resultSet.getObject(key);
+                            detailsMap.put(key.toLowerCase(Locale.ENGLISH), value);
+                        }
+                        lastReadPollingColumnValue = resultSet.getInt(pollingColumn);
+                        handleEvent(detailsMap);
+                    }
+                } catch (SQLException ex) {
+                    log.error(ex);
+                } finally {
+                    CDCPollingUtil.cleanupConnection(resultSet, null, null);
+                }
+                try {
+                    if (breakOnMissingRecord) {
+                        breakOnMissingRecord = false;
+                    }
+                    Thread.sleep((long) pollingInterval * 1000);
+                } catch (InterruptedException e) {
+                    log.error("Error while polling. Current mode: " + CDCSourceConstants.MODE_POLLING, e);
```

Comment:
Please improve the errors with the stream and App name information... else it will be hard to troubleshoot. please fix this for all cases.

Reply:
Fixed in b221509.